docs/servo_power_monitor/src/graph.ts

```diff
@@ -36,6 +47,34 @@ export class Graph {
       false
     );
   }
+  public addAnnotation(x: number, text: string) {
+    function capitalize(str: string) {
+      return str.charAt(0).toUpperCase() + str.slice(1).toLowerCase();
+    }
+    const capitalizedText = capitalize(text);
+    const newAnnotation = {
+      series: 'ina0',
+      x: x,
+      shortText: capitalizedText,
+      text: capitalizedText,
+      width: 36,
+      cssClass: 'annotation',
+    };
+    this.annotations.push(newAnnotation);
+    this.g.setAnnotations(this.annotations);
+  }
+  public findAnnotationPoint(
+    powerData: Array<PowerData>,
+    time: number,
+    text: string
+  ) {
+    for (const powerDataElement of powerData) {
```

Comment:
Maybe walking over the elements in a reversed order and finding the first element that is earlier than the annotation point would be faster than this logic